godtools/App/Features/Menu/Presentation/SocialSignIn/SocialSignInViewModel.swift

```diff
@@ -60,26 +60,30 @@ class SocialSignInViewModel: ObservableObject {
     }
     
     private func authenticateUser(provider: AuthenticationProviderType) {
-                        
-        authenticateUserUseCase.authenticatePublisher(provider: provider, policy: .renewAccessTokenElseAskUserToAuthenticate(fromViewController: presentAuthViewController))
-            .receiveOnMain()
-            .sink { [weak self] subscriberCompletion in
-                
-                let authenticationError: Error?
-                
-                switch subscriberCompletion {
-                case .finished:
-                    authenticationError = nil
-                case .failure(let error):
-                    authenticationError = error
-                }
-                
-                self?.handleAuthenticationCompleted(error: authenticationError)
-                
-            } receiveValue: { _ in
-
+        
+        authenticateUserUseCase.authenticatePublisher(
+            provider: provider,
+            policy: .renewAccessTokenElseAskUserToAuthenticate(fromViewController: presentAuthViewController),
+            createUser: authenticationType == .createAccount
+        )
+        .receiveOnMain()
+        .sink { [weak self] subscriberCompletion in
+            
+            let authenticationError: Error?
+            
+            switch subscriberCompletion {
+            case .finished:
+                authenticationError = nil
+            case .failure(let error):
+                authenticationError = error
             }
-            .store(in: &cancellables)
+            
+            self?.handleAuthenticationCompleted(error: authenticationError)
```

Comment:
Hey @rachaelblue looked into this crash a little more and I think I see what's going on, I'm just not entirely sure on a fix for it.

Our custom ```receiveOnMain()``` will successfully switch to the main thread for Self.Output using the flatMap.  However, since this crash happens from a thrown Error, flatMap won't get called in our custom ```receiveOnMain()```. 

```
public extension Publisher {
    
    func receiveOnMain() -> AnyPublisher<Self.Output, Self.Failure> {
        
        return self.flatMap({ value -> AnyPublisher<Self.Output, Self.Failure> in
            
            if Thread.isMainThread {
                
                return Just(value)
                    .setFailureType(to: Self.Failure.self)
                    .eraseToAnyPublisher()
                
            } else {
                
                return Just(value)
                    .setFailureType(to: Self.Failure.self)
                    .receive(on: DispatchQueue.main)
                    .eraseToAnyPublisher()
                
            }
            
        }).eraseToAnyPublisher()
    }
}
```

Reply:
Went ahead and created this task here -> https://jira.cru.org/browse/GT-2052.